### PR TITLE
feat: Add CSV plugin

### PR DESCRIPTION
**Pull request by dmikolay**
### Proposed Change
Added a CSV plugin.

##### Checklist
- [X] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: plugins/csv_logs.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    default: utf-8
+  - name: log_type
+    type: string
+    default: csv
+  - name: start_at
+    type: string
+    supported:
+    - beginning
+    - end
+    default: end
```
Comment (cpheps) [inline plugins/csv_logs.yaml:38]: `start_at` is in here twice. It looks like it is in the original too.

**Review by cpheps — COMMENTED** [review]
Overall looks good, did you test this against any log files to verify?

File: plugins/csv_logs.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      start_at: {{ .start_at }}
+      {{ if .exclude_file_log_path }}
+      exclude:
+        {{ range $fp := .exclude_log_paths }}
+        - '{{ $fp }}'
+        {{end}}
+      {{end}}
+      {{ if .encoding }}
+      encoding: '{{ .encoding }}'
+      {{end}}
```
Comment (jsirianni) [inline plugins/csv_logs.yaml:55]: Encoding has a default, no need to wrap with an `if`.

```suggestion
      encoding: '{{ .encoding }}'
```

File: plugins/csv_logs.yaml (hunk 1 above)
Comment (dmikolay) [reply plugins/csv_logs.yaml]: I just removed one of the repeated start-at. I redid some testing and it appear to still work just fine.

File: plugins/csv_logs.yaml (hunk 2 above)
Comment (dmikolay) [reply plugins/csv_logs.yaml]: Just removed the if statement wrapping encoding, thanks!

File: plugins/csv_logs.yaml (hunk 1 above)
Comment (dmikolay) [reply plugins/csv_logs.yaml]: Also, I did test against some log files, it seemed to function correctly.

**Review by cpheps — APPROVED** [review]

**Review by StefanKurek — APPROVED** [review]

**Pull request merged**